plugins/autoid/plugin.js

```diff
@@ -103,10 +103,12 @@
 
         for (i = 0; i < pastedElements.length; i++) {
           element = pastedElements[i];
-          id = element.attributes.id;
-          originalHeading = checkForDuplicateId(id);
-          if (originalHeading) {
-            element = resolveDuplicateIds(element, originalHeading);
+          if (element.attributes) {
```

Comment:
Ah I realize now some of the initial confusion.

Really, pastedElements is pastedNodes.

Does it make more sense to check the node type instead?

http://docs.ckeditor.com/#!/api/CKEDITOR.htmlParser.element

None of the other node types have attributes though.  So this is functionally the same.